front/app/modules/commercial/smart_groups/components/UserFilterConditions/rules.ts

```diff
@@ -487,43 +515,43 @@ export const ruleTypeConstraints = {
     not_is_normal_user: null,
   },
   participated_in_project: {
-    in: ProjectValueSelector,
+    in: ProjectValuesSelector,
```

Comment:
Is it desired behavior that for the positive selections there is a multi select, while for negative selections (e.g. not_posted_in, not_commened_in, etc.) only 1 value can be selected?

Reply:
Yes, that's the requirement. For negation, multiple values do not make much sense semantically so we keep single values in those cases.

Reply:
The negatives are definitely not included in the `MULTIVALUE_PREDICATES`.

I seem to remember us discussing this, and I _think_ we felt that needing to select members of a smart group based on a predicate of not having done one of 'x OR y OR ...' was unlikely (though I can kind of imagine use cases), but I could be mis-remembering.

Reply:
> @sebastienhoorens @jinjagit See my question below.
> 
> The front-end components are okay, but they are not written in the way we do it now. Simon, I refactored one as an example. Would you perhaps like to have a go at the other two to get practice with it? ...

@brentguf @sebastienhoorens  I'd love to have a go! :+1:

Reply:
There would be two ways to interpret negation on multiple values:
- No participation in any of those projects. This is currently already possible by adding multiple rules (one for each value).
- You didn't participate in one of those projects. Could be useful, but I think there will be a very low demand for this. Currently the work around for this is to create multiple groups (one for each value).